### PR TITLE
Making main tile accessible to pipes/other storage mods

### PR DESCRIPTION
**Pull request by ajsnarr98**
I have written plenty of java code, but never written any part of a Minecraft mod before, so this probably needs a bit of review.

WARNING: NEEDS TESTING. I do not know the normal way you would compile and test this mod.

#283

### CODE REVIEW

File: src/main/java/com/lothrazar/storagenetwork/capability/handler/MasterItemStackHandler.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  }
+  
+  @Override
+  public int getSlotLimit(int slot)
+  {
+    return Integer.MAX_VALUE;
+  }
+  
+  @Override
+  protected void onLoad() {
```
Comment (ajsnarr98) [inline src/main/java/com/lothrazar/storagenetwork/capability/handler/MasterItemStackHandler.java:87]: I am not sure when this is called to be honest, so we may need to call update() in other places.

**Comment by ajsnarr98** [comment]
If this works, this should probably be changable in a config option for the mod, so modpack authors can turn it off if needed.

File: src/main/java/com/lothrazar/storagenetwork/capability/handler/MasterItemStackHandler.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+/**
+ * ItemStackHandler used for interfacing with TileMain like a chest.
+ */
+public class MasterItemStackHandler extends ItemStackHandlerEx {
+  
+  private TileMain tileMain;
+  
+  public MasterItemStackHandler(TileMain tileMain) {
+    this(tileMain, Integer.MAX_VALUE);
+  }
```
Comment (ajsnarr98) [inline src/main/java/com/lothrazar/storagenetwork/capability/handler/MasterItemStackHandler.java:20]: I just realized a problem with using Integer.MAX_VALUE. The list inside the ItemStackHandler itself should be non-null, so the entire list will be backed by an array of size Integer.MAX_VALUE.

I could try to make the "size" of the inventory dynamic, if this negatively impacts memory use in any testing @Lothrazar can do, but also I don't know if changing the size of this inventory would have negative consequences. Could just use a smaller, big number. Or whatever the actual stack limit in the network currently is.

**Comment by Lothrazar** [comment]
Thanks for this. What mods are you testing this with? such as refined storage or ae2?

**Comment by Lothrazar** [comment]
@ajsnarr98 I think i have a version of this working in 1.16 . connect with me on twitter or discord (7738) if you want to chat more. Ill merge it soon, so you get credit at least

**Comment by Lothrazar** [comment]
Going to create standalone blocks that act as the StackHandler interfaces instead of the main tile

**Pull request merged**